pike/test/__init__.py

```diff
@@ -40,73 +40,358 @@
-    @staticmethod
-    def booloption(name, default='no'):
-        table = {'yes': True, 'no': False, '': False}
-        return table[PikeTest.option(name, 'no')]
+    @classmethod
+    def booloption(cls, name, default='no'):
+        table = {'yes': True, 'true': True, 'no': False, 'false': False, '': False}
+        return table[cls.option(name, default=default).lower()]
 
-    @staticmethod
-    def smb2constoption(name, default=None):
-        return getattr(smb2, PikeTest.option(name, '').upper(), default)
+    @classmethod
+    def smb2constoption(cls, name, default=None):
+        return getattr(smb2, cls.option(name, '').upper(), default)
+
+    @classmethod
+    def loglevel(cls):
+        return getattr(logging, cls.option(cls.PIKE_LOGLEVEL, default='NOTSET').upper())
+
+    @classmethod
+    def trace(cls):
+        return cls.booloption(cls.PIKE_TRACE)
+
+    @classmethod
+    def server(cls):
+        return cls.option(cls.PIKE_SERVER, default=_Required)
+
+    @classmethod
+    def port(cls):
+        return int(cls.option(cls.PIKE_PORT, default='445'))
+
+    @classmethod
+    def creds(cls):
+        return cls.option(cls.PIKE_CREDS)
+
+    @classmethod
+    def share(cls):
+        return cls.option(cls.PIKE_SHARE, "c$")
+
+    @classmethod
+    def signing(cls):
+        return cls.booloption(cls.PIKE_SIGN)
+
+    @classmethod
+    def encryption(cls):
+        return cls.booloption(cls.PIKE_ENCRYPT)
+
+    @classmethod
+    def min_dialect(cls):
+        return cls.smb2constoption(cls.PIKE_MIN_DIALECT, default=0)
+
+    @classmethod
+    def max_dialect(cls):
+        return cls.smb2constoption(cls.PIKE_MAX_DIALECT, default=float('inf'))
+
+
+def default_client():
+    client = model.Client()
+    min_dialect = Options.min_dialect()
+    max_dialect = Options.max_dialect()
+    client.dialects = [d for d in client.dialects
+                       if min_dialect <= d <= max_dialect]
+    if Options.signing():
+        client.security_mode = (smb2.SMB2_NEGOTIATE_SIGNING_ENABLED |
+                                smb2.SMB2_NEGOTIATE_SIGNING_REQUIRED)
+    return client
+
+
+@attr.s
+class TreeConnect(object):
+    """
+    Combines a Client, Connection, Channel, and Tree for simple access to an SMB share.
+    """
+    _client = attr.ib(default=None)
+    server = attr.ib(factory=Options.server)
+    port = attr.ib(factory=Options.port)
+    creds = attr.ib(factory=Options.creds)
+    share = attr.ib(factory=Options.share)
+    resume = attr.ib(default=None)
+    encryption = attr.ib(factory=Options.encryption)
+    require_dialect = attr.ib(default=None)
+    require_capabilities = attr.ib(default=None)
+    require_share_capabilities = attr.ib(default=None)
+    conn = attr.ib(default=None, init=False)
+    chan = attr.ib(default=None, init=False)
+    tree = attr.ib(default=None, init=False)
+
+    @require_dialect.validator
+    def _require_dialect_validator(self, attribute, value):
+        if value and len(value) != 2:
```

Comment:
Looks like this can accept a list. Are we expecting only tuples type? I don't think so. The below line saying tuple with syntax confuses me.

Reply:
`require_dialect` could accept any sequence type with length 2, only the verbiage suggests using a tuple.

I suppose it could be changed to "must be specified as a sequence with 2 elements [min_dialect, max_dialect]", but I don't know how much of a gain that would be.

Reply:
It is fine. I am ok.